grype/version/kb_contraint.go

```diff
@@ -37,14 +37,10 @@ func (c kbConstraint) supported(format Format) bool {
 }
 
 func (c kbConstraint) Satisfied(version *Version) (bool, error) {
-	if c.raw == "" && version != nil {
-		// an empty constraint is always satisfied
-		return true, nil
+	if c.raw == "" {
+		// an empty constraint is never satisfied
```

Comment:
@wagoodman What are your thoughts here? My understanding is that this is inconsistent with how version constraints work everywhere else in Grype, and if so, I'm concerned this might be confusing to Grype devs.

Reply:
Whoa — https://github.com/anchore/grype/pull/421#discussion_r711279804